src/main/java/microsoft/exchange/webservices/data/core/request/ServiceRequestBase.java

```diff
@@ -632,16 +638,28 @@ protected SoapFaultDetails readSoapFault(EwsServiceXmlReader reader) {
   protected HttpWebRequest validateAndEmitRequest() throws Exception {
     this.validate();
 
-    HttpWebRequest request = this.buildEwsHttpWebRequest();
+    final HttpWebRequest request = this.buildEwsHttpWebRequest();
 
     try {
+      ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
This is no good, and will leak memory. Create a single (static, most likely) ExecutorService, perhaps in a static constructor for this class. Also, you need to shutdown and cleanup the executor. You might be able to use the javax annotation '@PreDestroy' or you may have to expose some static methods that ptr-mail has to call to clean up.

Reply:
Added shutdown and moved executor service into a singleton